src/overlays/actors/ovl_En_Ik/z_en_ik.c

```diff
@@ -132,115 +224,871 @@ static DamageTable D_8092C140 = {
+        this->colliderCylinder.base.acFlags &= ~AC_HIT;
+    } else if ((this->colliderCylinder.base.acFlags & AC_HIT)) {
+        sp2C = 0;
+        Actor_SetDropFlag(&this->actor, &this->colliderCylinder.info);
+        this->colliderCylinder.base.acFlags &= ~AC_HIT;
+        if ((this->actor.colChkInfo.damageEffect != 15) &&
+            ((this->drawEffect != 10) || ((this->colliderCylinder.info.acHitInfo->toucher.dmgFlags & 0xDB0B3) == 0))) {
+            Actor_SetColorFilter(&this->actor, 0x4000, 255, 0, 12);
+            this->invincibilityFrames = 12;
+            EnIk_Thaw(this, globalCtx);
+            this->colliderQuad.base.atFlags &= ~AT_ON;
+            if (Actor_ApplyDamage(&this->actor) == 0) {
+                if (this->drawArmorFlags == 0) {
+                    this->actor.colChkInfo.health = 9;
+                    sp2C = 1;
+                } else {
+                    Enemy_StartFinishingBlow(globalCtx, &this->actor);
+                    func_801A2ED8();
+                }
+            }
+            if (sp2C == 1) {
+                this->drawArmorFlags = 1;
+                this->colliderCylinder.base.colType = 3;
+                this->actor.colChkInfo.damageTable = &sDamageTableNoArmor;
+                Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_IRONNACK_ARMOR_OFF_DEMO);
+                EnIk_SetupCutscene(this);
+            } else if (this->drawArmorFlags) {
+                if (this->actor.colChkInfo.damageEffect == 3) {
+                    EnIk_Freeze(this);
+                    EnIk_SetupFrozen(this);
+                    if (this->actor.colChkInfo.health == 0) {
+                        this->counter = 3;
+                        this->invincibilityFrames = 12;
+                    }
+                } else {
+                    if (this->actor.colChkInfo.damageEffect == 2) {
+                        this->effectAlpha = 4.0f;
+                        this->effectScale = 0.65f;
+                        this->drawEffect = 0;
+                    } else if (this->actor.colChkInfo.damageEffect == 4) {
+                        EnIk_HitArmor(this, globalCtx);
+                    }
+                    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_IRONNACK_DAMAGE);
+                    if (this->actor.colChkInfo.health != 0) {
+                        EnIk_SetupReactToAttack(this, 1);
+                    } else {
+                        EnIk_SetupCutscene(this);
+                    }
+                }
+            } else {
+                Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_IRONNACK_ARMOR_HIT);
+                Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_IRONNACK_DAMAGE);
+                if (this->actor.colChkInfo.damageEffect == 4) {
+                    EnIk_HitArmor(this, globalCtx);
+                }
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/func_80929BEC.s")
+void EnIk_ArmorDestroyed(EnIk* this, GlobalContext* globalCtx) {
```

Comment:
This seems to run unconditionally in Update, so isn't it just to do with updating the armour in general, or what?

Reply:
Would you say `EnIk_UpdateArmor` would make more sense? I think my original thoughts for Destroy were because it uses `SoundSource_PlaySfxAtFixedWorldPos` when the armor is destroyed.